### PR TITLE
Refine JavaScript Language Page

### LINKED ISSUE #1884: Improve JS/Node docs with example for private dependencies
## Is your feature request related to a problem? Please describe
When I first tried to setup Gitpod for my JS (React / Node) projects, I struggled in finding examples showing how to install private dependencies from NPM.

## Describe the behaviour you'd like
When I follow the tutorial [Node/TypeScript/Express Quickstart](https://www.gitpod.io/docs/quickstart/typescript), in the "Next Steps" section, I want to see an example on how to install my private NPM dependencies (e.g. add the secret `NPM_TOKEN`, customize your Dockerfile, etc)

## Describe alternatives you've considered
Provide a GitHub repo template with setup for private dependencies

## Additional context
N/A

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
Refine the whole JavaScript Language Page & will also fix #1884

### [**`Live Preview`**](https://deploy-preview-2965--gitpod-kumquat.netlify.app/docs/introduction/languages/javascript)

---

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2965"></a>

### CODE REVIEW

**Comment by Siddhant-K-code** [comment]
FYI: @Cahllagerfeld I've started we can work on this parallel (according the availability)

File: gitpod/docs/introduction/languages/javascript.md [hunk 1]
```diff
@@ -77,9 +79,37 @@ and then search for `eslint` in the extensions tab and then install it from ther
...
+
+```js title="vite.config.js"
+const config = () => ({
+  server: {
+    hmr: {
+      clientPort: process.env.HMR_HOST ? 443 : 5173,
+      host: process.env.HMR_HOST
+        ? new URL(process.env.HMR_HOST).hostname
+        : "localhost",
+    },
```
Comment (Cahllagerfeld) [inline gitpod/docs/introduction/languages/javascript.md:107]: This codesnippet explains how to configure HMR to work with Gitpod and localhost. Do you think we should remove the localhost part? @Siddhant-K-code

File: gitpod/docs/introduction/languages/javascript.md [hunk 2]
```diff
@@ -9,21 +9,23 @@ title: JavaScript in Gitpod
...
-| [Tesseract.js](https://github.com/naptha/tesseract.js)       | Pure JavaScript OCR for more than 100 Languages                                         | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/naptha/tesseract.js)       |
-| [freeCodeCamp](https://github.com/freeCodeCamp/freeCodeCamp) | [freeCodeCamp.org](https://www.freecodecamp.org/)'s open source codebase and curriculum | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/freeCodeCamp/freeCodeCamp) |
-| [Mozilla PDF.js](https://github.com/mozilla/pdf.js)          | PDF.js is a PDF viewer that is built with HTML5                                         | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/mozilla/pdf.js)            |
+## Getting started / Quick Start
 
-</div>
+`youtube: w65POyu3ZUQ`
+
+To see a full working Node/TypeScript application, take a look at [gitpod-io/template-typescript-node](https://github.com/gitpod-io/template-typescript-node). To update an existing JavaScript application, follow the steps below in this guide.
```
Comment (Siddhant-K-code) [inline gitpod/docs/introduction/languages/javascript.md:24]:
```suggestion
To see a full working Node/TypeScript application, take a look at [gitpod-samples/template-typescript-node](https://github.com/gitpod-samples/template-typescript-node). To update an existing JavaScript application, follow the steps below in this guide.
```

File: gitpod/docs/introduction/languages/javascript.md [hunk 3]
```diff
@@ -9,21 +9,23 @@ title: JavaScript in Gitpod
...
+
+<a href="https://gitpod.io/#https://github.com/gitpod-io/template-typescript-node">
```
Comment (Siddhant-K-code) [inline gitpod/docs/introduction/languages/javascript.md:26]:
```suggestion
<a href="https://gitpod.io/#https://github.com/gitpod-samples/template-typescript-node">
```

File: gitpod/docs/introduction/languages/javascript.md [hunk 4]
```diff
@@ -55,16 +82,107 @@ RUN bash -c 'VERSION="14.8.0" \
 RUN echo "nvm use default &>/dev/null" >> ~/.bashrc.d/51-nvm-fix
 ```
 
-and then in your [.gitpod.yml](/docs/references/gitpod-yml) reference your [.gitpod.Dockerfile](/docs/configure/workspaces/workspace-image) as shown below:
+5. Commit and push both `gitpod.yml` and `.gitpod.Dockerfile`
```
Comment (Siddhant-K-code) [inline gitpod/docs/introduction/languages/javascript.md:85]:
```suggestion
5. Commit and push both `.gitpod.yml` and `.gitpod.Dockerfile`
```

**Pull request merged**